src/PackageProject.cmake

```diff
@@ -0,0 +1,155 @@
+# Uses ycm (permissive BSD-3-Clause license) and FowardArguments (permissive MIT license)
+
+macro(package_project)
+  set(_options # default to true
+      ARCH_INDEPENDENT)
+  set(_oneValueArgs
+      # default all of these to the project_name or the given name:
+      NAME
```

Comment:
I think defaulting this to project_name could lead to confusing errors for future users. I find very rarely is a project simple enough that the project_name is also the exact name of the target you want to export.

Reply:
I strongly believe that the name of the project should be the same as the name of the package. Otherwise, the project is just confusing everyone. For `TARGETS`, sometimes, multiple targets are needed to be exported. So, I think we can require the `TARGETS` property, but not the `NAME`. In an ideal world, the name of the project should be the namespace, prefix, etc, but the additional target names can be different.

Note that these options are related to [ycm's options](https://github.com/robotology/ycm/blob/1860bad63dcdf6727c1ce0746561e1cada0dccef/modules/InstallBasicPackageFiles.cmake#L17). I removed a lot of them that cause more pain than gains.

Reply:
I see, I thought NAME was the name of one given target.

Reply:
Modified in 
https://github.com/cpp-best-practices/project_options/pull/58/commits/769883e641f331ba0af7f3a18dc163823057b767
and
https://github.com/cpp-best-practices/project_options/pull/58/commits/26f45ae5370543d558a6e8de1ddf8bd7115da117